### PR TITLE
Support for ghc9, th-2.17

### PR DESCRIPTION
**Pull request by odr**
In TH 2.17 quotation return `Quote m => m bla` instead of `Q bla`. It was the main problem

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -2,7 +2,10 @@ nix:
...
-resolver: lts-10.0
+resolver: lts-19.6
 extra-deps:
-- dom-parser-3.0.0
-- generic-arbitrary-0.1.0
+- git: https://github.com/typeable/dom-parser.git
+  commit: 7d556ba528bfa216e99714f8a0d4d00f452803a9
+
+- git: https://github.com/fumieval/xml-lens.git
+  commit: 4612e5ea18822baef678d7176c1ab5f93e4e31e9
```
Comment (ilyakooo0) [inline stack.yaml]: maybe we should release them on hackage instead?

File: src/Data/THGen/XML/Internal.hs [hunk 2]
```diff
@@ -160,18 +160,18 @@ instance Description ExhaustivenessName IsoXmlDescEnum where
   exhaustivenessName =:= (IsoXmlDescEnum genType descEnumCons) =
     isoXmlGenerateEnum genType exhaustivenessName (reverse descEnumCons)
 
-instance IsString (TH.TypeQ -> IsoXmlDescPreField) where
+instance (m ~ Q) => IsString (m TH.Type -> IsoXmlDescPreField) where
```
Comment (ilyakooo0) [inline src/Data/THGen/XML/Internal.hs:163]: Will this not work?

```suggestion
instance IsString (Q TH.Type -> IsoXmlDescPreField) where
```

File: stack.yaml (hunk 1 above)
Comment (odr) [reply stack.yaml]: I've got them from our projects. Not sure if they are needed really here. I will check.
(Anyway we can release only dom-parser I suppose)

File: src/Data/THGen/XML/Internal.hs (hunk 2 above)
Comment (odr) [reply src/Data/THGen/XML/Internal.hs:163]: > Will this not work?

It was in the last version. And it is not working with TH-2.17. I've added note for PR at the top.

File: stack.yaml (hunk 1 above)
Comment (odr) [reply stack.yaml]: I've cleanup it a bit

**Review by Yuras — APPROVED** [review]

**Pull request merged**